hatchet/readers/timemory_reader.py

```diff
@@ -236,13 +237,14 @@ def parse_node(_key, _dict, _hparent, _rank):
             _rank_dict = _keys if self.per_rank else _extra
 
             # handle the rank
-            _rank_dict["rank"] = collapse_ids(_rank)
+            _rank_dict["rank"] = int(collapse_ids(_rank))
```

Comment:
So I think you might want to condition this depending on whether multiple processes are in the output. If you look at the comment for the `collapse_ids` function you will see there was a reason I made it a string:

```python
        def collapse_ids(_obj):
            """node/rank/thread id may be int, array of ints, or None.
            This function generates a consistent form which is not numerical to
            avoid groupby(...).sum() from producing something nonsensical but
            the entry is still hashable
            """
```

So unless there was a fix in hatchet to `groupby(...).sum()` not adding together the rank IDs, it should be something like:

```python
            _rank_dict["rank"] = int(collapse_ids(_rank)) if self.per_rank else collapse_ids(_rank)
```

but I am going to change the `collapse_ids` function to perform the appropriate conversion.

```suggestion
            _rank_dict["rank"] = collapse_ids(_rank, self.per_rank)
```